pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -215,6 +218,26 @@ private void extractFromTableConfig(TableConfig tableConfig) {
     }
   }
 
+  /**
+   * Extracts compressionType for each column. Populates a map containing column name as key and compression type as
+   * value. Note that only RAW forward index columns will be populated in this map.
```

Comment:
I'd recommend adding a short comment here stating that this map only contains the overrides for the compression type and does not include the default value for METRICS (PASSTHROUGH) or DIMENSION (LZ4) no dictionary columns.

Perhaps add a note about `getColumnCompressionType()` in `SegmentColumnarIndexCreater` setting up the defaults if the `fieldConfig` doesn't contain the compression codec.

Just want to ensure that in the future no one uses this assuming that all compression types are extracted and defaults are filled in for the compression types.

Reply:
Done.